flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-jdbc/src/main/java/org/apache/flink/cdc/connectors/jdbc/sink/v2/JdbcWriter.java

```diff
@@ -0,0 +1,248 @@
+                    getUpsertStatementBuilder(rowData.getSchema().getColumns());
+
+            JdbcOutputFormat<Object, JdbcRowData, JdbcBatchStatementExecutor<JdbcRowData>>
+                    outputFormat =
+                            jdbcUpsertOutputs.computeIfAbsent(
+                                    rowData.getTableId(),
+                                    id -> getJdbcOutputFormat(upsertStmt, upsertStmtBuilder));
+
+            outputFormat.writeRecord(rowData);
+            outputFormat.flush();
+            LOG.debug(
+                    "Successfully upserted into table `{}`.`{}`. Raw SQL: {}",
+                    rowData.getTableId().getSchemaName(),
+                    rowData.getTableId().getTableName(),
+                    upsertStmt);
+        }
+
+        // delete event
+        if (RowKind.DELETE.is(rowData.getRowKind())) {
+            String deleteStmt =
+                    catalog.getDeleteStatement(
+                            rowData.getTableId(), rowData.getSchema().primaryKeys());
+            JdbcStatementBuilder<JdbcRowData> delStmtBuilder =
+                    getDeleteStatementBuilder(rowData.getSchema().primaryKeys());
+
+            JdbcOutputFormat<Object, JdbcRowData, JdbcBatchStatementExecutor<JdbcRowData>>
+                    outputFormat =
+                            jdbcDeleteOutputs.computeIfAbsent(
+                                    rowData.getTableId(),
+                                    id -> getJdbcOutputFormat(deleteStmt, delStmtBuilder));
+
+            outputFormat.writeRecord(rowData);
+            outputFormat.flush();
+            LOG.debug(
+                    "Successfully deleted rows in table `{}`.`{}`. Raw SQL: {}",
+                    rowData.getTableId().getSchemaName(),
+                    rowData.getTableId().getTableName(),
+                    deleteStmt);
+        }
+    }
+
+    private JdbcOutputFormat<Object, JdbcRowData, JdbcBatchStatementExecutor<JdbcRowData>>
+            getJdbcOutputFormat(String sql, JdbcStatementBuilder<JdbcRowData> builder) {
+        JdbcOutputFormat<Object, JdbcRowData, JdbcBatchStatementExecutor<JdbcRowData>>
+                jdbcOutputFormat =
+                        new JdbcOutputFormat<>(
+                                connectionProvider,
+                                executionOptions,
+                                () -> JdbcBatchStatementExecutor.simple(sql, builder));
+        try {
+            jdbcOutputFormat.open(outputSerializer);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return jdbcOutputFormat;
+    }
+
+    @Override
+    public void flush(boolean b) throws IOException, RuntimeException {
+        // To ensure the sequential execution of insert and delete,
+        // PrepareStatements cannot perform insert and delete batch operations at the same time,
+        // so flush operations are not used separately.
```

Comment:
I think there may be performance issues. 
However, this depends on us emitting information on whether it is in the snapshot reading stage, so we can handle it uniformly in the future.

Reply:
Refactored `JdbcWriter` and introduced `write.batch.size` option to improve performance.